qiskit_addon_obp/backpropagation.py

```diff
@@ -164,7 +164,8 @@ def handle_timeout(signum, frame):
             # PERF: we will likely need to parallelize this loop
             for i in range(num_observables):
                 non_trivial_slice = False
-                for op_idx, op_node in enumerate(circuit_to_dag(slice_).topological_op_nodes()):
+                op_nodes = reversed(list(circuit_to_dag(slice_).topological_op_nodes()))
+                for op_idx, op_node in enumerate(op_nodes):
```

Comment:
Since this `op_idx` now indexes the gates in the layer in reverse, I wonder whether we should update the log message where this gets used. Just to be super clear about this.

Reply:
Done

Reply:
Actually you do need to cast the above line to a list. `reversed` doesn't work on generators